ansible/roles/load-balancer/templates/haproxy.cfg.j2

```diff
@@ -0,0 +1,40 @@
+#
+# {{ ansible_managed }}
+#
+global
+    log /dev/log    local0
+    log /dev/log    local1 notice
+    chroot /var/lib/haproxy
+    user haproxy
+    group haproxy
+    daemon
+    stats socket /var/lib/haproxy/stats level admin
+
+defaults
+    log global
+    option  dontlognull
+    timeout connect 5000
+    timeout client 50000
+    timeout server 50000
+
+frontend localhost
+    bind *:{{ kube_master_frontend_api_port }}
```

Comment:
Instead of using * in line 21, I believe you need to specify the vip:

```{{ kube_apiserver_vip }}```

I believe if you do not specify kube_apiserver_vip, then haproxy will listen on all addresses.